todo-list/src/api/todo.ts

```diff
@@ -0,0 +1,31 @@
+import { Todo } from "../types/todos/todos.type";
+import { safeFetch } from "./safeFetch";
+
+export async function fetchTodosByUserId(userId: number): Promise<Todo[]> {
+  const url = `/todos/user/${userId}`;
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error("Failed to fetch todo");
+  }
+  const todo: Todo[] = await response.json();

```

Comment:
`response.json()` can throw an error if the response's body is not in JSON format.

Add a try catch here to avoid unexpected errors